ciphers/rsa_factorization.py

```diff
@@ -30,25 +28,33 @@ def rsafactor(d: int, e: int, N: int) -> list[int]:
     [113, 241]
     >>> rsafactor(4021, 13, 17711)
     [89, 199]
+    >>> rsafactor(int("123c5b61ba36edb1d3679904199a89ea80c09b9122e1400c09adcf77846" +
+    ...    "76d01d23356a7d44d6bd8bd50e94bfc723fa87d8862b75177691c11d757692df8881", 16),
+    ...    0x10001,
+    ...    int("a66791dc6988168de7ab77419bb7fb0c001c62710270075142942e19a8d8c51d05" +
+    ...    "3b3e3782a1de5dc5af4ebe99468170114a1dfe67cdc9a9af55d655620bbab", 16))[0]
+    23234950162188993388155927630085331316851060055334470382368804331834850828939
+    >>> rsafactor(4, 3, 13)
+    Traceback (most recent call last):
+    ...
+    ValueError: e and d are not valid RSA exponents
     """
     k = d * e - 1
-    p = 0
-    q = 0
-    while p == 0:
+    if k & 1 == 1:
+        raise ValueError("e and d are not valid RSA exponents")
+
+    while True:
         g = random.randint(2, N - 1)
         t = k
-        while True:
-            if t % 2 == 0:
-                t = t // 2
-                x = (g ** t) % N
-                y = math.gcd(x - 1, N)
-                if x > 1 and y > 1:
-                    p = y
-                    q = N // y
-                    break  # find the correct factors
-            else:
-                break  # t is not divisible by 2, break and choose another g
-    return sorted([p, q])
+        while t & 1 == 0:
+            t >>= 1
+            x = pow(g, t, N)
+            y = math.gcd(x - 1, N)
+            if x > 1 and y > 1:
+                # we found the correct factors
+                p = y
+                q = N // y
+                return sorted([p, q])

```

Comment:
```suggestion
                p = y
                q = N // y
                return tuple(sorted([p, q]))
```
Convert output to a tuple (see comment above)